app/routes/api/applications.js

```diff
@@ -192,6 +193,115 @@ export const applicationHandlers = [{
 
       await updateApplicationData(reference, updatedProperty, newValue, oldValue, note, user)
 
+      return h.response().code(204)
+    }
+  }
+},
+{
+  method: 'post',
+  path: '/api/application/{ref}/flag',
+  options: {
+    validate: {
+      params: joi.object({
+        ref: joi.string().valid()
+      }),
+      payload: joi.object({
+        user: joi.string().required(),
+        note: joi.string().required(),
+        appliesToMh: joi.bool().required()
+      }),
+      failAction: async (request, h, err) => {
+        request.logger.setBindings({ err })
+        return h.response({ err }).code(400).takeover()
+      }
+    },
+    handler: async (request, h) => {
+      const { user, note, appliesToMh } = request.payload
+      const { ref } = request.params
+
+      request.logger.setBindings({ appliesToMh, user, note, ref })
+
+      const application = await findApplication(ref)
+
+      if (application === null) {
+        return h.response('Not Found').code(404).takeover()
+      }
+
+      const flag = await getFlagByAppRef(ref)
```

Comment:
Not sure this works. If they already had 2 flags on application, and for whatever reason tried to add a 3rd, if the getFlagByAppRef happens to fetch just the one with the opposite appliesToMh flag, we'd then let them create that 3rd flag, but that's not our intent

Reply:
Good point yeah - I'll change this so that you also pass in the appliesToMh